design/data-flow.md

```diff
@@ -0,0 +1,68 @@
+        - Calls `CheckTransaction` for each transaction, defined at: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L1056
+            - is coinbase
+            - calls `CheckTransactionWithoutProofVerification`, defined at: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L1078
+                - 'Basic checks that don't require any context'
+                - With verifier == false, it looks like in this flow
+                - Checks a whole bunch of stuff that we check with our semantic representation into structs
+                    - We should double check that there are no conditions on the values and amounts that we aren't already checking 
+                        - Check for duplicate 
+                            - inputs
+                            - nullifiers (within a single transaction)
+                            - // Transactions containing empty `vin` must have either non-empty `vJoinSplit` or non-empty `vShieldedSpend`.
+                            - // Transactions containing empty `vout` must have either non-empty `vJoinSplit` or non-empty `vShieldedOutput`.
+                            - Moar: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L1091
+        - Sum up "LegacySigOps" for each transaction and check that it's less than some maximum 
+    - Acquires a lock, then calls `MarkBlockAsReceived` (networking?)
+    - Calls `AcceptBlock`, defined at: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L4180
+        - Calls `AcceptBlockHeader`, defined at: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L4130
+            - Also calls `CheckBlockHeader`
+            - Calls `ContextualCheckBlockHeader`, defined at: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L3900
+                - Does checks given a pointer to the previous block
+                - Check Equihash solution is valid
+                    - In our code we compute the equihash solution on the block alone, we will need to also do a step to check that its block height is the appopriate N+1 re: the previous block
+                - Check proof of work
+                - Check timestamp against prev
+                - Check future timestamp soft fork rule introduced in v2.1.1-1.
+                    - https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L4027
+                - Check timestamp
+                    - https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L4040
+                - Reject block.nVersion < 4 blocks
+                - Don't accept any forks from the main chain prior to last checkpoint
+                    - We will probably get this 'more naturally' and don't need to explicitly check it like this
```

Comment:
CheckpointVerifier effectively does this, and when we don't have hard-coded checkpoints, the parallel verification RFC should handle it #763